src/Handlebars/Template.php

```diff
@@ -129,14 +129,17 @@ public function render($context)
         $topTree = end($this->stack); // never pop a value from stack
         list($index, $tree, $stop) = $topTree;
 
+        // Whitespace control
+        $mustStripWhitespaceBefore = false;
+
         $buffer = '';
         while (array_key_exists($index, $tree)) {
             $current = $tree[$index];
-            $index++;
```

Comment:
This was moved to lower in the `while` block to keep `$index` pointed at the current item.